### ISSUE TITLE
SQLModel codegen produces syntax errors

### ISSUE THREAD

**Issue by NixBiks** [agronholm/sqlacodegen#302]
### Things to check first

- [X] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release

### Sqlacodegen version

3.0.0rc3

### SQLAlchemy version

2.0.23

### RDBMS vendor

PostgreSQL

### What happened?

The screenshot shows some of the issues.

1. There are unused imports (can easily be fixed by running ruff after codegen)
2. Missing imports, i.e. `mapped_column`.
3. Syntax error due to a `,` in line 22.

<img width="1001" alt="Screenshot 2023-11-16 at 09 18 05" src="https://github.com/agronholm/sqlacodegen/assets/16841792/0512f107-0c51-4de9-878d-e86014d715b3">

### Database schema for reproducing the bug

```sql
CREATE TABLE
  logs (
    timestamp timestamp NOT NULL,
    level text NOT NULL,
    message text NOT NULL,
    schedule_id bigint REFERENCES schedules (id),
    worker_id text,
    run_id text,
    id bigint GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY
  );

-- sort by timestamp
CREATE INDEX logs_timestamp_desc_idx ON logs (timestamp DESC);

-- create view of logs
CREATE VIEW logs_view AS
  SELECT
    timestamp,
    level,
    message,
    schedule_id,
    worker_id,
    run_id
  FROM logs
  ORDER BY timestamp DESC;
```

**Comment by madsenwattiq** [comment]
I see this as well.  In order to fix broken code, I had to manually add the "metadata" variable to all association tables that didn't have primary keys - that's the bug with the double commas.  After that, it works in 3.0.0rc3.

**Comment by agronholm** [comment]
It's not supposed to use `mapped_column` at all. All the tests expect that it uses `Column()` and not `mapped_column()`. The latter is a thing only on SQLAlchemy 2. Before `sqlmodel` 0.0.12 (released 7 hours ago), it didn't even work with SQLAlchemy 2 at all.

**Comment by michelmetran** [comment]
> I'm pretty sure that this was caused by `sqlmodel now supporting SQLAlchemy 2, and sqlacodegen wasn't ready for that.

Thank you for the information!

After that, I used an environment with SQLAlchemy < 2 and it worked!

```
conda create --name sqlacodegen-py39 -c conda-forge python=3.9 sqlacodegen==3.0.0rc3 sqlalchemy==1.4.51 pymssql==2.2.7
```

<br>

Now the mapped_column no longer appears, as it is not yet compatible with SQLModel

**Comment by yaoguoba** [comment]
This issue still exists in rc5, which still includes mapped_column. I replaced mapped_column with sqlmodel.Column， You can work normally now.

**Comment by yaoguoba** [comment]
This issue still exists in rc5, which still includes mapped_column. I replaced mapped_column with sqlmodel.Column， You can work normally now.

**Comment by superpan** [comment]
I am running 3.0.0rc5 with sqlalchemy 2.0.36, and it's still producing `mapped_column`.  Should I manually replace all mapped_columns with sqlmodel.Column?

**Comment by agronholm** [comment]
> I am running 3.0.0rc5 with sqlalchemy 2.0.36, and it's still producing `mapped_column`. Should I manually replace all mapped_columns with sqlmodel.Column?

Yes. I haven't been able to focus on this project, and nobody has volunteered to take on the maintainership, so I don't expect progress very soon.

**Comment by sheinbergon** [comment]
@agronholm is this the ticket?

**Comment by agronholm** [comment]
One of them at least. I've added one more to the milestone.

**Comment by sheinbergon** [comment]
Great. Feel free to assign this one to me :smile:

**Comment by sheinbergon** [comment]
Great. I'll update in 2 or 3 weeks :rabbit:

**Comment by sheinbergon** [comment]
@agronholm trying to make good on my word -  here's the PR - https://github.com/agronholm/sqlacodegen/pull/358

From what I tested no, more codegen issues.  Let me know what do you think, and I'll move to fix more tests

**Comment by agronholm** [comment]
Thanks, I'll try to review it during the weekend!
